pkg/epp/handlers/request.go

```diff
@@ -32,6 +32,37 @@ import (
 func (s *StreamingServer) HandleRequestHeaders(reqCtx *RequestContext, req *extProcPb.ProcessingRequest_RequestHeaders) error {
 	reqCtx.RequestReceivedTimestamp = time.Now()
 
+	// Headers must be processed first to populate the request context as subsequent logic (like body processing) may
+	// depend upon it.
+	for _, header := range req.RequestHeaders.Headers.Headers {
+		key := header.Key
+		// Per the Envoy API, a header's value can be in either the `RawValue` (bytes) or `Value` (string) field.
+		if header.RawValue != nil {
+			reqCtx.Request.Headers[key] = string(header.RawValue)
+		} else {
+			reqCtx.Request.Headers[key] = header.Value
+		}
```

Comment:
continuing @ahg-g line of thinking - I agree we should fix the `ExtractHeaderValue` helper func and then we can use it here, e.g.,:
```go
reqCtx.Request.Headers[key] = ExtractHeaderValue(key).
```
I think the logic that handles RawValue or Value should be scoped to a single helper function.